### ISSUE TITLE
[BUG] meshing compilation fails in some environments

### ISSUE THREAD

**Issue by barentine** [python-microscopy/python-microscopy#478]
**Describe the bug**
```
PYME/experimental/_triangle_mesh.c:11004:3: warning: passing argument 4 of ‘_update_face_normals’ from incompatible pointer type [[0/327]
 by default]
   _update_face_normals((&(*((__pyx_t_5numpy_int32_t *) ( /* dim=0 */ (__pyx_v_idxs.data + __pyx_t_7 * __pyx_v_idxs.strides[0]) )))), __$
yx_v_self->_chalfedges, __pyx_v_self->_cvertices, __pyx_v_self->_cfaces, (__pyx_v_idxs.shape[0]));
   ^
In file included from PYME/experimental/_triangle_mesh.c:633:0:
PYME/experimental/triangle_mesh_utils.c:316:13: note: expected ‘struct face_t *’ but argument is of type ‘struct face_d *’
 static void _update_face_normals(int32_t *f_idxs, halfedge_t *halfedges, vertex_t *vertices, face_t *faces, signed int n_idxs)
             ^
error: Command "gcc -pthread -B /home/ubuntu/miniconda3/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Ws$
rict-prototypes -fPIC -IPYME/experimental -I/home/ubuntu/miniconda3/lib/python3.6/site-packages/numpy/core/include -I/home/ubuntu/minicon
da3/lib/python3.6/site-packages/numpy/core/include -I/home/ubuntu/miniconda3/include/python3.6m -I/home/ubuntu/miniconda3/include/python3
.6m -c PYME/experimental/_triangle_mesh.c -o build/temp.linux-x86_64-3.6/PYME/experimental/_triangle_mesh.o -MMD -MF build/temp.linux-x86
_64-3.6/PYME/experimental/_triangle_mesh.o.d -O3 -fno-exceptions -ffast-math -march=native -mtune=native" failed with exit status 1

```
<details> <summary> on this build </summary>

```
# packages in environment at /home/ubuntu/miniconda3:
#
# Name                    Version                   Build  Channel
absl-py                   0.9.0                    pypi_0    pypi
alabaster                 0.7.12                   py36_0    anaconda
appdirs                   1.4.3                    pypi_0    pypi
asn1crypto                1.3.0                    py36_0    anaconda
astor                     0.8.1                    pypi_0    pypi
attrs                     19.3.0                     py_0    anaconda
babel                     2.8.0                      py_0    anaconda
beautifulsoup4            4.9.1                    py36_0    anaconda
blas                      1.0                         mkl    anaconda
brotlipy                  0.7.0           py36h7b6447c_1000    anaconda
bzip2                     1.0.8                h7b6447c_0    anaconda
ca-certificates           2020.7.22                     0    anaconda
cairo                     1.14.12              h8948797_3    anaconda
certifi                   2020.6.20                py36_0    anaconda
cffi                      1.14.0           py36h2e261b9_0    anaconda
chardet                   3.0.4                 py36_1003    anaconda
cheroot                   8.3.0                    py36_0    anaconda
cherrypy                  18.1.1                   py36_0    anaconda
cloudpickle               1.5.0                      py_0    anaconda
conda                     4.8.4                    py36_0    anaconda
conda-build               3.18.11                  py36_0    anaconda
conda-package-handling    1.6.1            py36h7b6447c_0    anaconda
contextlib2               0.6.0.post1                py_0    anaconda
cryptography              2.9.2            py36h1ba5d50_0    anaconda
cycler                    0.10.0                   py36_0    anaconda
cython                    0.29.13          py36he6710b0_0    anaconda
cytoolz                   0.10.1           py36h7b6447c_0    anaconda
dask-core                 2.20.0                     py_0    anaconda
dbus                      1.13.12              h746ee38_0    anaconda
decorator                 4.4.2                      py_0    anaconda
dispatch                  1.0                      py36_0    david_baddeley
docutils                  0.16                     py36_1    anaconda
expat                     2.2.9                he6710b0_2    anaconda
fftw                      3.3.8                h7b6447c_3    anaconda
filelock                  3.0.12                     py_0    anaconda
fontconfig                2.13.0               h9420a91_0    anaconda
freetype                  2.10.2               h5ab3b9f_0    anaconda
fribidi                   1.0.9                h7b6447c_0    anaconda
future                    0.18.2                   py36_1    anaconda
gast                      0.3.3                    pypi_0    pypi
gettext                   0.19.8.1             h9b4dc7a_1    anaconda
glib                      2.56.2               hd408876_0    anaconda
glob2                     0.7                        py_0    anaconda
graphite2                 1.3.14               h23475e2_0    anaconda
grpcio                    1.31.0                   pypi_0    pypi
gst-plugins-base          1.14.0               hbbd80ab_1    anaconda
gstreamer                 1.14.0               hb453b48_1    anaconda
h5py                      2.10.0                   pypi_0    pypi
harfbuzz                  1.8.8                hffaf4a1_0    anaconda
hdf5                      1.10.1               hb0523eb_0    anaconda
icu                       58.2                 he6710b0_3    anaconda
idna                      2.10                       py_0    anaconda
ifaddr                    0.1.6                      py_0    david_baddeley
imageio                   2.9.0                      py_0    anaconda
imagesize                 1.2.0                      py_0    anaconda
importlib-metadata        1.7.0                    py36_0    anaconda
importlib_metadata        1.7.0                         0    anaconda
intel-openmp              2020.1                      217    anaconda
jaraco.functools          3.0.1                      py_0    anaconda
jinja2                    2.11.2                     py_0    anaconda
joblib                    0.16.0                     py_0    anaconda
jpeg                      9b                   habf39ab_1    anaconda
keras                     2.2.4                    pypi_0    pypi
keras-applications        1.0.8                    pypi_0    pypi
keras-preprocessing       1.1.2                    pypi_0    pypi
kiwisolver                1.2.0            py36hfd86e86_0    anaconda
lcms2                     2.11                 h396b838_0    anaconda
libarchive                3.4.2                h62408e4_0    anaconda
libedit                   3.1.20191231         h14c3975_1    anaconda
libffi                    3.2.1                h4deb6c0_3    anaconda
libgcc-ng                 9.1.0                hdf63c60_0    anaconda
libgfortran-ng            7.3.0                hdf63c60_0    anaconda
libglu                    9.0.0                hf484d3e_1    anaconda
liblief                   0.10.1               he6710b0_0    anaconda
libpng                    1.6.37               hbc83047_0    anaconda
libstdcxx-ng              9.1.0                hdf63c60_0    anaconda
libtiff                   4.1.0                h2733197_1    anaconda
libuuid                   1.0.3                h1bed415_2    anaconda
libxcb                    1.14                 h7b6447c_0    anaconda
libxml2                   2.9.10               he19cac6_1    anaconda
localization-survey       0.0                       dev_0    <develop>
lz4-c                     1.9.2                he6710b0_1    anaconda
lzo                       2.10                 h7b6447c_2    anaconda
mako                      1.1.0                    pypi_0    pypi
markdown                  3.2.2                    pypi_0    pypi
markupsafe                1.1.1            py36h7b6447c_0    anaconda
mask-rcnn                 2.1                       dev_0    <develop>
matplotlib                3.2.2                         0    anaconda
matplotlib-base           3.2.2            py36hef1b27d_0    anaconda
mkl                       2019.5                      281    anaconda
mkl-service               2.3.0            py36he904b0f_0    anaconda
mkl_fft                   1.1.0            py36h23d657b_0    anaconda
mkl_random                1.1.0            py36hd6b4f25_0    anaconda
more-itertools            8.4.0                      py_0    anaconda
mpld3                     0.3                      py36_0    anaconda
ncurses                   6.2                  he6710b0_1    anaconda
networkx                  2.4                        py_1    anaconda
numexpr                   2.7.1            py36h423224d_0    anaconda
numpy                     1.14.6           py36h3b04361_4    anaconda
numpy-base                1.14.6           py36h81de0dd_4    anaconda
nvidia-ml-py3             7.352.0                  pypi_0    pypi
olefile                   0.46                     py36_0    anaconda
omero-py                  5.7.1                      py_1    ome
openssl                   1.1.1                h7b6447c_0    anaconda
packaging                 20.4                       py_0    anaconda
pandas                    0.25.1           py36he6710b0_0    anaconda
pango                     1.42.4               h049681c_0    anaconda
patchelf                  0.11                 he6710b0_0    anaconda
pcre                      8.44                 he6710b0_0    anaconda
pillow                    7.2.0            py36hb39fc2d_0    anaconda
pip                       20.1.1                   py36_1    anaconda
pixman                    0.40.0               h7b6447c_0    anaconda
pkginfo                   1.5.0.1                  py36_0    anaconda
pluggy                    0.13.1                   py36_0    anaconda
portend                   2.5                        py_0    anaconda
protobuf                  3.12.4                   pypi_0    pypi
psutil                    5.7.0            py36h7b6447c_0    anaconda
py                        1.9.0                      py_0    anaconda
py-lief                   0.10.1           py36h403a769_0    anaconda
pycairo                   1.19.1           py36h2a1e443_0    anaconda
pycosat                   0.6.3            py36h7b6447c_0    anaconda
pycparser                 2.20                       py_2    anaconda
pycuda                    2019.1.2                 pypi_0    pypi
pyface                    7.0.1                    py36_1    anaconda
pyfftw                    v0.11.1          py36h39e3cac_0    david_baddeley
pygments                  2.6.1                      py_0    anaconda
pyme-depends              1.08                     py36_0    local
pyme-omero                20.8.25                   dev_0    <develop>
pyme-rcnn                 0                         dev_0    <develop>
pyme-warp-drive           0.3                       dev_0    <develop>
pymecompress              0.2.0                    py36_0    local
pyopengl                  3.1.1a1                  py36_0    anaconda
pyopenssl                 19.1.0                     py_1    anaconda
pyparsing                 2.4.7                      py_0    anaconda
pyqt                      5.9.2            py36h22d08a2_1    anaconda
pysocks                   1.7.1                    py36_0    anaconda
pytables                  3.4.2            py36h3b5282a_2    anaconda
pytest                    5.3.1                    py36_0    anaconda
python                    3.6.9                h265db76_0  
python-dateutil           2.8.1                      py_0    anaconda
python-libarchive-c       2.9                        py_0    anaconda
python-microscopy         20.10.5                   dev_0    <develop>
pytools                   2019.1.1                 pypi_0    pypi
pytz                      2020.1                     py_0    anaconda
pywavelets                1.1.1            py36h7b6447c_0    anaconda
pyyaml                    5.3.1            py36h7b6447c_1    anaconda
qt                        5.9.7                h5867ecd_1    anaconda
readline                  7.0                  h7b6447c_5    anaconda
repoze.lru                0.7                      py36_0    anaconda
requests                  2.24.0                     py_0    anaconda
ripgrep                   11.0.2               he32d670_0    anaconda
routes                    2.4.1                      py_1    anaconda
ruamel_yaml               0.15.87          py36h7b6447c_1    anaconda
scikit-image              0.15.0           py36he6710b0_0    anaconda
scikit-learn              0.23.1           py36h423224d_0    anaconda
scipy                     1.5.0            py36h0b6359f_0    anaconda
setuptools                49.2.0                   py36_0    anaconda
simplejson                3.17.0           py36h7b6447c_0    anaconda
sip                       4.19.23          py36he6710b0_0    anaconda
six                       1.15.0                     py_0    anaconda
snowballstemmer           2.0.0                      py_0    anaconda
soupsieve                 2.0.1                      py_0    anaconda
sphinx                    3.1.2                      py_0    anaconda
sphinxcontrib-applehelp   1.0.2                      py_0    anaconda
sphinxcontrib-devhelp     1.0.2                      py_0    anaconda
sphinxcontrib-htmlhelp    1.0.3                      py_0    anaconda
sphinxcontrib-jsmath      1.0.1                      py_0    anaconda
sphinxcontrib-qthelp      1.0.3                      py_0    anaconda
sphinxcontrib-serializinghtml 1.1.4                      py_0    anaconda
sqlite                    3.32.3               h62c20be_0    anaconda
tempora                   3.0.0                      py_0    anaconda
tensorboard               1.12.2                   pypi_0    pypi
tensorflow                1.12.0                   pypi_0    pypi
tensorflow-gpu            1.12.0                   pypi_0    pypi
termcolor                 1.1.0                    pypi_0    pypi
threadpoolctl             2.1.0              pyh5ca1d4c_0    anaconda
tk                        8.6.10               hbc83047_0    anaconda
toolz                     0.10.0                     py_0    anaconda
toposort                  1.1                      py36_2    david_baddeley
tornado                   6.0.4            py36h7b6447c_1    anaconda
tqdm                      4.47.0                     py_0    anaconda
traits                    5.1.1            py36h7b6447c_0    anaconda
traitsui                  7.0.1                      py_0    anaconda
ujson                     3.0.0            py36he6710b0_0    anaconda
urllib3                   1.25.9                     py_0    anaconda
wcwidth                   0.2.5                      py_0    anaconda
werkzeug                  1.0.1                    pypi_0    pypi
wheel                     0.34.2                   py36_0    anaconda
wxpython                  4.0.4            py36hc99224d_0    anaconda
xz                        5.2.5                h7b6447c_0    anaconda
yaml                      0.2.5                h7b6447c_0    anaconda
zc.lockfile               2.0                        py_0    anaconda
zeroc-ice36-python        3.6.5            py36h6bb024c_6    ome
zeroconf                  0.23.0                   py36_3    david_baddeley
zipp                      3.1.0                      py_0    anaconda
zlib                      1.2.11               h7b6447c_3    anaconda
zstd                      1.4.4                h0b5b093_3    anaconda

```

</details>

<details> <summary> while runs successfully under this one </summary>

```
# packages in environment at /home/ubuntu/miniconda3:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main  
alabaster                 0.7.12                   pypi_0    pypi
appdirs                   1.4.4                    pypi_0    pypi
astroid                   2.4.2                    py37_0    anaconda
attrs                     19.3.0                     py_0    anaconda
babel                     2.8.0                    pypi_0    pypi
blas                      1.0                         mkl  
brotlipy                  0.7.0           py37h7b6447c_1000    anaconda
bzip2                     1.0.8                h7b6447c_0    anaconda
ca-certificates           2020.7.22                     0    anaconda
cairo                     1.14.12              h8948797_3  
certifi                   2020.6.20                py37_0    anaconda
cffi                      1.14.0           py37he30daa8_1    anaconda
chardet                   3.0.4                 py37_1003    anaconda
cheroot                   8.4.2                    pypi_0    pypi
cherrypy                  18.6.0                   pypi_0    pypi
conda                     4.8.4                    py37_0    anaconda
conda-package-handling    1.6.1            py37h7b6447c_0    anaconda
cryptography              2.9.2            py37h1ba5d50_0    anaconda
cycler                    0.10.0                   pypi_0    pypi
cython                    0.29.21          py37he6710b0_0    anaconda
decorator                 4.4.2                    pypi_0    pypi
dispatcher                1.0                      pypi_0    pypi
django                    1.11.29                  pypi_0    pypi
docutils                  0.16                     pypi_0    pypi
enum-compat               0.0.3                    pypi_0    pypi
expat                     2.2.9                he6710b0_2    anaconda
fontconfig                2.13.0               h9420a91_0    anaconda
freetype                  2.10.2               h5ab3b9f_0    anaconda
fribidi                   1.0.9                h7b6447c_0    anaconda
future                    0.18.2                   py37_1    anaconda
gettext                   0.19.8.1             h9b4dc7a_1    anaconda
glib                      2.65.0               h3eb4bd4_0    anaconda
graphite2                 1.3.14               h23475e2_0    anaconda
gst-plugins-base          1.14.0               hbbd80ab_1  
gstreamer                 1.14.0               hb31296c_0  
harfbuzz                  2.4.0                hca77d97_1    anaconda
icu                       58.2                 he6710b0_3    anaconda
idna                      2.9                        py_1  
imageio                   2.9.0                    pypi_0    pypi
imagesize                 1.2.0                    pypi_0    pypi
importlib-metadata        1.7.0                    py37_0    anaconda
importlib_metadata        1.7.0                         0    anaconda
intel-openmp              2020.1                      217  
isort                     4.3.21                   py37_0    anaconda
jaraco-classes            3.1.0                    pypi_0    pypi
jaraco-collections        3.0.0                    pypi_0    pypi
jaraco-functools          3.0.1                    pypi_0    pypi
jaraco-text               3.2.0                    pypi_0    pypi
jinja2                    2.11.2                   pypi_0    pypi
joblib                    0.16.0                   pypi_0    pypi
jpeg                      9b                   habf39ab_1    anaconda
kiwisolver                1.2.0                    pypi_0    pypi
lazy-object-proxy         1.4.3            py37h7b6447c_0    anaconda
lcms2                     2.11                 h396b838_0    anaconda
ld_impl_linux-64          2.33.1               h53a641e_7  
libedit                   3.1.20181209         hc058e9b_0  
libffi                    3.3                  he6710b0_1  
libgcc-ng                 9.1.0                hdf63c60_0  
libgfortran-ng            7.3.0                hdf63c60_0  
libglu                    9.0.0                hf484d3e_1    anaconda
libpng                    1.6.37               hbc83047_0    anaconda
libstdcxx-ng              9.1.0                hdf63c60_0  
libtiff                   4.1.0                h2733197_1    anaconda
libuuid                   1.0.3                h1bed415_2    anaconda
libxcb                    1.14                 h7b6447c_0    anaconda
libxml2                   2.9.10               he19cac6_1    anaconda
lz4-c                     1.9.2                he6710b0_1    anaconda
mako                      1.1.3                    pypi_0    pypi
markupsafe                1.1.1                    pypi_0    pypi
matplotlib                3.3.0                    pypi_0    pypi
mccabe                    0.6.1                    py37_1    anaconda
mkl                       2020.1                      217  
mkl-service               2.3.0            py37he904b0f_0  
mkl_fft                   1.1.0            py37h23d657b_0  
mkl_random                1.1.1            py37h0573a6f_0    anaconda
more-itertools            8.4.0                      py_0    anaconda
mpld3                     0.5.1                    pypi_0    pypi
ncurses                   6.2                  he6710b0_1  
netifaces                 0.10.9                   pypi_0    pypi
networkx                  2.4                      pypi_0    pypi
numexpr                   2.7.1                    pypi_0    pypi
numpy                     1.19.1           py37hbc911f0_0  
numpy-base                1.19.1           py37hfa32c7d_0  
nvidia-ml-py3             7.352.0                  pypi_0    pypi
olefile                   0.46                     py37_0    anaconda
omero-py                  5.7.1                      py_1    ome
openssl                   1.1.1g               h7b6447c_0    anaconda
packaging                 20.4                       py_0    anaconda
pandas                    1.1.0                    pypi_0    pypi
pango                     1.45.3               hd140c19_0    anaconda
pcre                      8.44                 he6710b0_0    anaconda
pillow                    7.2.0            py37hb39fc2d_0    anaconda
pip                       20.1.1                   py37_1    anaconda
pixman                    0.40.0               h7b6447c_0    anaconda
pluggy                    0.13.1                   py37_0    anaconda
portend                   2.6                      pypi_0    pypi
psutil                    5.7.2                    pypi_0    pypi
py                        1.9.0                      py_0    anaconda
pycosat                   0.6.3            py37h7b6447c_0    anaconda
pycparser                 2.20                       py_0  
pycuda                    2019.1.2                 pypi_0    pypi
pyface                    7.0.1                    pypi_0    pypi
pyfftw                    0.12.0                   pypi_0    pypi
pygments                  2.6.1                    pypi_0    pypi
pylint                    2.5.3                    py37_0    anaconda
pyme-omero                20.8.25                   dev_0    <develop>
pyme-warp-drive           0.3                       dev_0    <develop>
pymecompress              0.2.1                    pypi_0    pypi
pyopengl                  3.1.5                    pypi_0    pypi
pyopenssl                 19.1.0                     py_1    anaconda
pyparsing                 2.4.7                      py_0    anaconda
pysocks                   1.7.1                    py37_1    anaconda
pytest                    5.4.3                    py37_0    anaconda
python                    3.7.7                hcff3b4d_5    anaconda
python-dateutil           2.8.1                    pypi_0    pypi
python-microscopy         20.10.5                   dev_0    <develop>
pytools                   2020.3.1                 pypi_0    pypi
pytz                      2020.1                   pypi_0    pypi
pywavelets                1.1.1                    pypi_0    pypi
pyyaml                    5.3.1            py37h7b6447c_0    anaconda
readline                  8.0                  h7b6447c_0  
requests                  2.24.0                     py_0    anaconda
ruamel_yaml               0.15.87          py37h7b6447c_0    anaconda
scikit-image              0.17.2                   pypi_0    pypi
scikit-learn              0.23.2                   pypi_0    pypi
scipy                     1.5.0            py37h0b6359f_0  
setuptools                49.2.0                   py37_0    anaconda
six                       1.15.0                     py_0    anaconda
snowballstemmer           2.0.0                    pypi_0    pypi
sphinx                    3.1.2                    pypi_0    pypi
sphinxcontrib-applehelp   1.0.2                    pypi_0    pypi
sphinxcontrib-devhelp     1.0.2                    pypi_0    pypi
sphinxcontrib-htmlhelp    1.0.3                    pypi_0    pypi
sphinxcontrib-jsmath      1.0.1                    pypi_0    pypi
sphinxcontrib-qthelp      1.0.3                    pypi_0    pypi
sphinxcontrib-serializinghtml 1.1.4                    pypi_0    pypi
sqlite                    3.31.1               h62c20be_1  
tables                    3.6.1                    pypi_0    pypi
tempora                   4.0.0                    pypi_0    pypi
threadpoolctl             2.1.0                    pypi_0    pypi
tifffile                  2020.7.24                pypi_0    pypi
tk                        8.6.10               hbc83047_0    anaconda
toml                      0.10.1                     py_0    anaconda
toposort                  1.5                      pypi_0    pypi
tqdm                      4.46.0                     py_0  
traits                    6.1.1                    pypi_0    pypi
traitsui                  7.0.1                    pypi_0    pypi
typed-ast                 1.4.1            py37h7b6447c_0    anaconda
ujson                     3.1.0                    pypi_0    pypi
urllib3                   1.25.9                     py_0    anaconda
wcwidth                   0.2.5                      py_0    anaconda
wheel                     0.34.2                   py37_0    anaconda
wrapt                     1.11.2           py37h7b6447c_0    anaconda
wxpython                  4.0.4            py37hc99224d_0  
xz                        5.2.5                h7b6447c_0  
yaml                      0.1.7                h96e3832_1    anaconda
zc-lockfile               2.0                      pypi_0    pypi
zeroc-ice36-python        3.6.5            py37h6bb024c_6    ome
zeroconf                  0.17.7                   pypi_0    pypi
zipp                      3.1.0                      py_0    anaconda
zlib                      1.2.11               h7b6447c_3  
zstd                      1.4.4                h0b5b093_3    anaconda

```

</details>

**To Reproduce**
1. `python setup.py develop` in the 'wrong' environment

**Expected behavior**
egg link code without bricking

 - OS: [e.g. Win10] ubuntu 14.4 (fail)/ 18.04 (ok)
 - Python version: 3.6 (fail) / 3.7 (ok)
 - numpy version: 1.14 (fail) / 1.19 (ok)

**Comment by David-Baddeley** [comment]
From the environments, I suspect it has something to do with the cython version and the c code that cython is generating. It could also be something in the interaction between cython and numpy. 

It looks like cython versions 0.29.13 and 0.29.21 respectively. Might be worth looking at the changelog between these versions (or at the very least seeing if we can trigger by just changing cython version rather than the full environment)

**Comment by barentine** [comment]
can try updating cython to 0.29.21 shortly

**Comment by barentine** [comment]
![image](https://user-images.githubusercontent.com/31105780/95387288-81d59a00-08be-11eb-86d4-3f99f1f37c75.png)
cython version alone isn't enough.

**Comment by barentine** [comment]
For reference, gcc versions are 4.8.4 (not compiling) and 7.5.0 (compiling). Not that it will matter after #489

**Comment by barentine** [comment]
Still failing despite #489 @David-Baddeley and @zacsimile 
![image](https://user-images.githubusercontent.com/31105780/95614349-570d5200-0a34-11eb-9e94-ae664db90023.png)

**Comment by zacsimile** [comment]
This seems like a separate error from what was fixed in #489. This code was written about a year ago, so I'm surprised you haven't run into this compile bug before. Is there anything else new, package-wise?

**Comment by zacsimile** [comment]
O sorry, I was looking at the warning. The error appears to come from the `int` declaration in the `for` loop. Will upload a fix.

**Comment by barentine** [comment]
solved with #503 and #489. Thanks @David-Baddeley and @zacsimile
